src/components/map/CreateGroupCard.tsx

```diff
@@ -86,14 +77,40 @@ const CreateGroupCard = ({ onClose }: CreateGroupCardProps) => {
 
   // Creates a group with the selected users
   const handleCreateGroup = () => {
-    Alert.alert(
-      'TODO: create a group with these users: ',
-      JSON.stringify(
-        selectedUsers.map(user => user.firstName + ' ' + user.lastName)
-      )
-    );
-    // TODO: close if group creation is successful; otherwise, don't
-    onClose();
+    if (!userDocument) return;
+    if (selectedUsers.length === 0) return Alert.alert('No users selected!');
+
+    // create a group object
+    const groupObject: Group = {
+      name: 'Group_' + Math.floor(Math.random() * 1000),
+      members: [userDocument._id],
+      invitedMembers: selectedUsers.map(user => user._id),
+      creationDatetime: new Date(),
+      expirationDatetime: new Date(),
+    };
+
+    console.log('Attempting to create group...', groupObject);
+
+    // send a POST request to the server to create the group
+    fetch(`${SERVER_URL}groups?userId=${userDocument?._id}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(groupObject),
+    })
+      .then(res => res.json())
+      .then(data => {
+        // if response has an error message, display to user
+        if (data.message) Alert.alert(data.message);
+        else {
+          // otherwise, display success and close card
+          console.log('HERE IS THE GROUP', data);
+          Alert.alert('Group created successfully!');
+        }
+        updateUserDocument({});
```

Comment:
Can you explain why there is a {} in this? 

Comments!!

Reply:
yeah, the API for `updateUserDocument` already has comments for this

```ts
/**
   * Update the userDocument state.
   * Uses mongoDB id by default.
   * Optionally, pass in a firebaseUid to fetch via firebaseUid instead of mongoDB id.
   * @param firebaseUid - the firebase uid of the user
   */
  const updateUserDocument = async ({
    firebaseUid,
  }: UpdateUserDocumentInterface) => { ... }
```